Lacrit/assignment3/Main.hpp

```diff
@@ -0,0 +1,125 @@
+#include <map>
+#include <string>
+#include <unordered_set>
+#include <vector>
+#include "Dictionary.hpp"
+
+// prolly not the best solution to store it like that
+int rows = 2;
+int cols = 3;
+
+class Main {
+private:
+ 	struct cell {
+        std::string value;
+        int row;
+        int col;
+        std::vector<int> neighbours;
+        cell(int row, int col);
+    };	
+	std::vector<cell> board;
+    Dictionary dict;
+    std::unordered_set<std::string> found_words;
+    void Words(int pos, std::string str = "", std::unordered_set<int> visited = std::unordered_set<int>());
+
+public:
+    Main(Dictionary dict);
+    ~Main();
+
+    void Load(std::string letters);
+    void Print();
+	std::unordered_set<std::string> Result(); 
+};
+
+// Main board constructor
+Main::Main(Dictionary dict) 
+{
+	this->dict = dict;
+	int row = 0; 
+	int col = 0;
+
+	
+  for (int i = 0; i < rows; i++) {
+	  row = i / rows;
+	  for ( int j = 0; j < cols; j++) {
+        col = j % cols;
+        board.push_back(cell(row, col));
```

Comment:
(*)
I think here `row` is 0 and `col` is `j`.
You can use `i` and `j` directly, right?